include/ignition/gazebo/components/ReferenceModels.hh

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#ifndef IGNITION_GAZEBO_COMPONENTS_REFERENCEMODELS_HH_
+#define IGNITION_GAZEBO_COMPONENTS_REFERENCEMODELS_HH_
+
+#include <iostream>
+#include <set>
+
+#include <ignition/gazebo/Entity.hh>
+#include <ignition/gazebo/components/Factory.hh>
+#include <ignition/gazebo/components/Component.hh>
+#include <ignition/gazebo/config.hh>
+
+namespace ignition
+{
+namespace gazebo
+{
+// Inline bracket to help doxygen filtering.
+inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE {
+namespace components
+{
+  /// \brief Data structure that holds information about which models for
+  /// a given link view this link as the model's canonical link. An example
```

Comment:
You're right; this is specific to canonical links (there are no other intended use cases for this component at this time). Do you have any renaming suggestions? Perhaps `CanonicalLinkModels`?

Reply:
`CanonicalLinkModels` works. I don't have a strong opinion on the naming of this component - I was mainly thinking about the possibility of reusing this component for other purposes given that it has a generic name

Reply:
Another option would be to leave the name as `ReferenceModels` in case we do want to re-use this component for something else in the future. If we were to re-use this component without renaming it, all we'd have to do is change the documentation, which wouldn't break API/ABI. I don't have a strong preference either way - what option do you think is best?

Reply:
I've stored the canonical link mapping information in a class that's internal to the physics system, so we no longer need to worry about the case of changing a component that would break ABI: https://github.com/ignitionrobotics/ign-gazebo/pull/736#issuecomment-830184043